app/controllers/application_controller.rb

```diff
@@ -1,4 +1,24 @@
 class ApplicationController < ActionController::API
   include Response
   include ExceptionHandler
+
+  around_action :with_current_request
+
+  private
+
+  def with_current_request
+    ManageIQ::API::Common::Request.with_request(request) do |current|
+      begin
+        ActsAsTenant.with_tenant(current_tenant(current.user)) { yield }
+      rescue Exceptions::NoTenantError
+        json_response({ :message => 'Unauthorized' }, :unauthorized)
+      end
```

Comment:
@bzwei Is this missing an ensure to unset the thread local storage like here

https://github.com/ManageIQ/manageiq-api-common/blob/6d242a7fded6935557b769658480e364705ca6f9/lib/manageiq/api/common/request.rb#L30

Reply:
@bzwei 
ensure
  ActsAsTenant.current_tenant = nil

Reply:
Actually they are already doing this
https://www.rubydoc.info/gems/acts_as_tenant/0.3.9/ActsAsTenant.with_tenant